homeassistant/components/yamaha/yamaha_config_info.py

```diff
@@ -0,0 +1,37 @@
+"""Configuration Information for Yamaha."""
+
+from aiohttp import ClientSession
+import defusedxml.ElementTree as ET
+
+ns = {"s": "urn:schemas-upnp-org:device-1-0"}
+
+
+class YamahaConfigInfo:
+    """Configuration Info for Yamaha Receivers."""
+
+    def __init__(self, host: str) -> None:
+        """Initialize the Configuration Info for Yamaha Receiver."""
+        self.ctrl_url: str | None = f"http://{host}:80/YamahaRemoteControl/ctrl"
+
+    @classmethod
+    async def check_yamaha_ssdp(cls, location: str, client: ClientSession) -> bool:
+        """Check if the Yamaha receiver has a valid control URL."""
+        res = await client.get(location)
+        text = await res.text()
+        return (
+            text.find(
+                "<yamaha:X_controlURL>/YamahaRemoteControl/ctrl</yamaha:X_controlURL>"
+            )
+            != -1
+        )
+
+    @classmethod
+    async def get_upnp_serial_and_model(
+        cls, host: str, client: ClientSession
+    ) -> tuple[str, str]:
+        """Retrieve the serial_number and model from the SSDP description URL."""
+        res = await client.get(f"http://{host}:49154/MediaRenderer/desc.xml")
+        root = ET.fromstring(await res.text())
+        serial_number = root.find("./s:device/s:serialNumber", ns).text
```

Comment:
We should not do requests directly from the code, we want every integration to have their device specific code in a library.

Reply:
Alright, so how should I solve this then? Am I supposed to fork https://github.com/wuub/rxv and put it in there (since it's been unmaintained for a while?)

Reply:
Managed to reuse `rxv.ssdp.rxv_details` which handles this now.